### PR TITLE
CDNC-1781 Add restart command/api

### PR DESCRIPTION
**Pull request by allenchen2244**
Added an api and a cli command that restarts a previous workflow with the same parameters

Simple restart and terminate that's different from reset

Tested it locally

Nothing that bad can happen. Only caller is CLI.

### CODE REVIEW

File: tools/cli/app_test.go [hunk 1]
```diff
@@ -339,6 +339,20 @@ func (s *cliAppSuite) TestShowHistory_PrintDateTime() {
 	s.Nil(err)
 }
 
+func (s *cliAppSuite) TestRestartWorkflow() {
```
Comment (ZackLK) [inline tools/cli/app_test.go:342]: Are these used somewhere?

**Review by ZackLK — COMMENTED** [review]
Can you write a test for the RestartWorkflowExecution?

File: tools/cli/app_test.go (hunk 1 above)
Comment (allenchen2244) [reply tools/cli/app_test.go:342]: this is a unit test for the cli

Comment (ZackLK) [reply tools/cli/app_test.go:342]: Oh wow, not sure what I was thinking when I wrote that comment. They sure are unit tests for the cli. :D

**Comment by ZackLK** [comment]
> Can you write a test for the RestartWorkflowExecution?

Just to be a bit more clear about this comment of mine - it would be nice if there was a test that made sure that a Restart did what you expect it to do, that wasn't completely mocked out and exercised the new code you wrote.

**Comment by allenchen2244** [comment]
> Can you write a test for the RestartWorkflowExecution?

Test is added

**Review by davidporter-id-au — APPROVED** [review]

**Pull request merged**